### ISSUE TITLE
Making Editor Responsive.

### ISSUE THREAD

**Issue by NitinBhasneria** [publiclab/PublicLab.Editor#526]
Currently, the editor is not responsive for the small devices. This issue aims in making the editor responsive.

**Comment by jywarren** [comment]
Copying from #490 -- 

> @NitinBhasneria i believe the current approach is to use responsive and variably hidden/shown classes from Bootstrap to get the editor to work cleanly in both. However, please also take a look at @cesswairimu's work on a simple editor which we wanted to pilot alongside the main editor, for certain uses: #351 (and other related issues). Perhaps that could be adopted as the "mobile view" of the editor? It bears some discussion, what do people think?

**Comment by NitinBhasneria** [comment]
@jywarren Yea, this requires some discussion. Also, the points @shreyaa-sharmaa pointed are noticeable and should be solved.

**Comment by jywarren** [comment]
This might link to the convo with @Shulammite-Aso in #464 actually! Can you two sync up a bit about responsive classes in Bootstrap? Thank you so much!

**Comment by NitinBhasneria** [comment]
Things need to be fixed and the ways:
[ ] **Hovering the tips:** Tips take too much space in the editor module and so when the tip is not displayed the black space should be removed.
[ ] **Toolbar**: The present toolbar is very weird to look at the mobile screen. 
[ ] **Tips spacing:** The tips are taking more spaces than required which looks unsystematic in the mobile screen.
[ ] **The numbers left aligned and occupying a complete line:** This can be fixed by having one line in front of this.

We need some discussion on the mobile design of the toolbar @Shulammite-Aso @jywarren @keshav234156 @shreyaa-s. Please give your views.
![WhatsApp Image 2020-07-12 at 6 36 56 PM](https://user-images.githubusercontent.com/44112080/87247155-b37dc700-c46f-11ea-8f7d-7d46930c1e0d.jpeg)

**Comment by Shulammite-Aso** [comment]
> 1. Hovering the tips: Tips take too much space in the editor module and so when the tip is not displayed the black space should be removed.

@NitinBhasneria  I notice this too.
> 2. Toolbar: The present toolbar is very weird to look at the mobile screen.

This certainly needs to be worked on. There's a discussion about the possible ways to make the toolbar look instead going on here https://github.com/publiclab/plots2/issues/7067

> 3. Tips spacing: The tips are taking more spaces than required which looks unsystematic in the mobile screen.

i don't really get this part. You mind add a photo specifically for it?
> 4. The numbers left aligned and occupying a complete line: This can be fixed by having one line in front of this.

Also about this, do you mean having the numbering and help text on the same line?


You have eyes for details actually!!

**Comment by NitinBhasneria** [comment]
> > 1. Tips spacing: The tips are taking more spaces than required which looks unsystematic in the mobile screen.
> 
> i don't really get this part. You mind add a photo specifically for it?
> 

![hola](https://user-images.githubusercontent.com/44112080/87290003-0b253c80-c51b-11ea-899e-222fcbbf9209.jpeg)
You can see a lot of spaces between the lines than required.


> > 1. The numbers left aligned and occupying a complete line: This can be fixed by having one line in front of this.
> 
> Also about this, do you mean having the numbering and help text on the same line?

Yes,

**Comment by shreyaa-sharmaa** [comment]
How about adding a button that will open the detailed guidelines or something similar? It takes more space than the textarea, which seems a little strange. Also, it is only available when you click on the textarea and this feature looks neat in the desktop mode but not in mobile view in my opinion.

**Comment by NitinBhasneria** [comment]
> How about adding a button that will open the detailed guidelines or something similar? It takes more space than the textarea, which seems a little strange. Also, it is only available when you click on the textarea and this feature looks neat in the desktop mode but not in mobile view in my opinion.

I was thinking the guidelines(with space) will appear when we focus on particular module. What do you think about this

**Comment by shreyaa-sharmaa** [comment]
I think the guidelines would work the best if they guide a new user on the module use as well as not obstruct or hinder an experienced user in their work. What I wanted to highlight was a smooth transition from one module to to another. I'm not to sure of how this would look but if you think that guidelines on focusing would serve the purpose, I'm down with it. 
Also, that's a nice idea with the tooltips.

**Comment by NitinBhasneria** [comment]
Good idea @shreyaa-s. @jywarren @keshav234156 @Shulammite-Aso have a look at the comments and give your views.

**Comment by fiona-byte** [comment]
Hi, is this issue up for fix? I'd love to work on it

**Comment by Sagarpreet** [comment]
Yes @fiona-byte , this is still an enhancement we want. Feel free to work on this 😄

**Comment by fiona-byte** [comment]
Thank you @Sagarpreet 🙂

**Comment by Charu271** [comment]
Hi @NitinBhasneria 
Is this issue resolved ? If not I would like to work on it

**Comment by NitinBhasneria** [comment]
> Hi @NitinBhasneria
> Is this issue resolved ? If not I would like to work on it

Yea, there are still many works left in this issue. @fiona-byte If you are working on this please let us know. @Charu271 you can start the work if we get no reply from @fiona-byte

**Comment by Charu271** [comment]
Okay

**Comment by RaviAnand111** [comment]
Is this issue still open, I would like to work to make editor responsible.

**Comment by noobyogi0010** [comment]
Hi, @NitinBhasneria 
Is this issue still up for fixing? If so can I work upon it?

**Comment by sudipta-dotcom** [comment]
Is this issue still not resolved ? if not then I would be eager to resolve this one.
